### PR TITLE
[Keyboard] Add lxxt

### PR DESCRIPTION
**Pull request by DeskDaily**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/lxxt/info.json [hunk 1]
```diff
@@ -0,0 +1,130 @@
+{
+    "keyboard_name": "LXXT", 
+    "manufacturer": "DeskDaily",
+    "url": "", 
+    "maintainer": "DeskDaily", 
+    "usb": {
+    	"vid": "0x5003",
+    	"pid": "0x0001",
+    	"device_version": "0.0.1"
+    },
```
Comment (fauxpark) [inline keyboards/lxxt/info.json:10]:
```suggestion
    },
    "encoder": {
        "rotary": [
            {"pin_a": "GP27", "pin_b": "GP28"},
            {"pin_a": "GP8", "pin_b": "GP9"}
        ]
    },
```

File: keyboards/lxxt/config.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "config_common.h"
+
+#define ENCODERS 2
+#define ENCODERS_PAD_A { GP27, GP8 }
+#define ENCODERS_PAD_B { GP28, GP9 }
```
Comment (fauxpark) [inline keyboards/lxxt/config.h:23]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/lxxt/info.json [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        "bootmagic": true,
+        "mousekey": true,
+        "extrakey": true,
+        "console": false,
+        "command": false,
+        "nkro": true,
+        "backlight": false,
+        "rgblight": true,
+        "audio": false,
+        "key_lock": false
```
Comment (zvecr) [inline keyboards/lxxt/info.json:49]:
```suggestion
        "rgblight": true,
```

File: keyboards/lxxt/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS 
+    ),
+
+    [3] = LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS  
+    ),
```
Comment (zvecr) [inline keyboards/lxxt/keymaps/default/keymap.c:50]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by drashna — CHANGES_REQUESTED** [review]
Could you do this: 
https://docs.qmk.fm/#/feature_encoders?id=callbacks

**Review by fauxpark — APPROVED** [review]

File: keyboards/lxxt/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        }
+    } else if (index == 1) { /* Second encoder */
+        if (clockwise) {
+            tap_code_delay(KC_BRIU, 10);
+        } else {
+            tap_code_delay(KC_BRID, 10);
+        }
+    }
+    return false;
+}
```
Comment (zvecr) [inline keyboards/lxxt/keymaps/default/keymap.c:44]: Or encoder should be enabled by default
```suggestion
#if defined(ENCODER_ENABLE)
bool encoder_update_user(uint8_t index, bool clockwise) {
    if (index == 0) { /* First encoder */
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    } else if (index == 1) { /* Second encoder */
        if (clockwise) {
            tap_code_delay(KC_BRIU, 10);
        } else {
            tap_code_delay(KC_BRID, 10);
        }
    }
    return false;
}
#endif
```

**Review by drashna — APPROVED** [review]

File: keyboards/lxxt/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+#endif
```
Comment (drashna) [inline keyboards/lxxt/keymaps/default/keymap.c:48]: could this be moved to a lxxt.c file instead?

File: keyboards/lxxt/lxxt.h [hunk 7]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2023 DeskDaily
```
Comment (fauxpark) [inline keyboards/lxxt/lxxt.h:1]: This file can be deleted.

File: keyboards/lxxt/lxxt.c [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include "lxxt.h"
```
Comment (fauxpark) [inline keyboards/lxxt/lxxt.c:18]:
```suggestion
#include "quantum.h"
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/lxxt/keymaps/default/rules.mk
Comment (waffle87) [inline keyboards/lxxt/keymaps/default/rules.mk:1]: This file should be removed.

**Review by waffle87 — APPROVED** [review]

**Pull request merged**